pkg/snapshot/copier/copier.go

```diff
@@ -0,0 +1,128 @@
+	brtypes "github.com/gardener/etcd-backup-restore/pkg/types"
+
+	"github.com/sirupsen/logrus"
+)
+
+// GetSourceAndDestinationStores returns the source and destination stores for the given source and destination and store configs.
+func GetSourceAndDestinationStores(sourceSnapStoreConfig *brtypes.SnapstoreConfig, destSnapStoreConfig *brtypes.SnapstoreConfig) (brtypes.SnapStore, brtypes.SnapStore, error) {
+	sourceSnapStore, err := snapstore.GetSnapstore(sourceSnapStoreConfig)
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not get source snapstore: %v", err)
+	}
+
+	destSnapStore, err := snapstore.GetSnapstore(destSnapStoreConfig)
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not get destination snapstore: %v", err)
+	}
+
+	return sourceSnapStore, destSnapStore, nil
+}
+
+// Copier can be used to copy backups from a source to a destination store.
+type Copier struct {
+	logger           *logrus.Entry
+	sourceSnapStore  brtypes.SnapStore
+	destSnapStore    brtypes.SnapStore
+	maxSnapshots     int
+	lastSnapshotDays int
+}
+
+// NewCopier creates a new copier.
+func NewCopier(sourceSnapStore brtypes.SnapStore, destSnapStore brtypes.SnapStore, logger *logrus.Entry, maxSnapshots int, lastSnapshotDays int) *Copier {
+	return &Copier{
+		logger:           logger.WithField("actor", "copier"),
+		sourceSnapStore:  sourceSnapStore,
+		destSnapStore:    destSnapStore,
+		maxSnapshots:     maxSnapshots,
+		lastSnapshotDays: lastSnapshotDays,
+	}
+}
+
+// Run executes the copy command.
+func (c *Copier) Run() error {
+	return c.CopyBackups()
+}
+
+// CopyBackups copies all backups from the source store to the destination store.
+func (c *Copier) CopyBackups() error {
+	// Get source backups
+	c.logger.Info("Getting source backups...")
+	sourceBackups, err := c.getBackups()
+	if err != nil {
+		return fmt.Errorf("could not get source backups: %v", err)
+	}
+
+	// If there are no source backups, do nothing
+	if len(sourceBackups) == 0 {
+		c.logger.Info("No source backups found")
+		return nil
+	}
+
+	// Get destination snapshots and build a map keyed by name
+	c.logger.Info("Getting destination snapshots...")
+	destSnapshots, err := c.destSnapStore.List()
+	if err != nil {
+		return fmt.Errorf("could not get destination snapshots: %v", err)
+	}
+	destSnapshotsMap := make(map[string]*brtypes.Snapshot)
+	for _, snapshot := range destSnapshots {
+		destSnapshotsMap[snapshot.SnapName] = snapshot
+	}
+
+	for _, backup := range sourceBackups {
+		// Copy all the snapshots (if not already copied)
+		if _, ok := destSnapshotsMap[backup.SnapName]; !ok {
+			c.logger.Infof("Copying %s snapshot %s...", backup.Kind, backup.SnapName)
+			if err := c.copySnapshot(backup); err != nil {
+				return err
+			}
+		} else {
+			c.logger.Infof("Skipping %s snapshot %s as it already exists", backup.Kind, backup.SnapName)
+		}
+	}
+	return nil
+}
+
+func (c *Copier) getBackups() (brtypes.SnapList, error) {
+	if c.lastSnapshotDays > 0 {
```

Comment:
The default (meaning all) is -1, not 0.
```suggestion
	if c.maxBackupAge >= 0 {
```